configuration-file-v2.rst

```diff
@@ -0,0 +1,127 @@
+.. post:: January 30, 2019
+   :tags: feature
+   :author: Santos
+   :location: CUE
+
+New Configuration File
+======================
+
+We are happy to announce the new version of the `Read the Docs configuration file`_ (v2).
+
+.. _Read the Docs configuration file: https://docs.readthedocs.org/en/latest/config-file/v2
+
+.. code-block:: yaml
+
+   version: 2
+   python:
+     version: 3.7
+     install:
+       - requirements: docs/requirements.txt
+   sphinx:
+     configuration: docs/conf.py
+   formats: all
+
+If you are a recurrent Read the Docs user,
+chances are that you've configured your projects using a `.readthedocs.yml` file.
+
+Using the web interface is quick,
+but the main advantages of using a configuration file over the web interface are:
+
+- Some settings are only available using a configuration file.
+
+We are moving away from using the web interface.
+
+- The settings are per version rather than per project.
+   
+This would allow different versions of docs to be built with different versions of Python
+or different requirements.
+
+- The settings live in your VCS.
+
+- Reproducible build environments over time.
+
+What did we change in the new version?
+-------------------------------------
+
+- All valid options from the web interface are available.
+- We reorganized and rename some previous options to be more clear.
+- New defaults, we don't merge the values from the web interface.
+- Users can install their projects in a more flexible way.
+- Strict validation for invalid options,
+  to avoid typos and provide more feedback on invalid configurations.
+
+New features
+------------
+
+- MkDocs support (previously it was only supported from the web interface).
+
+.. code-block:: yaml
+
+   version: 2
+   mkdocs:
+     configuration: mkdocs.yml
+     fail_on_warning: true
+
+- More control over Sphinx.
+
+.. code-block:: yaml
+
+   version: 2
+   sphinx:
+      builder: html
+      configuration: docs/source/conf.py
+      fail_on_warning: true
+
+- Control over submodules, we don't always need them all to build our docs.
+
+.. code-block:: yaml
+
+   version: 2
+   submodules:
+     include:
+       - third-party/dependency
+     recursive: true
+
+- Support for multiple packages/requirements installations.
+
+.. code-block:: yaml
+
+   version: 2
+   python:
+     version: 3.7
+     install:
+       - requirements: docs/requirements.txt
+       - requirements: requirements.txt
+       - method: pip
+         path: package
+
+Future improvements
+-------------------
+
+- Pipfile support
+- Show the current configuration used in each build.
+- Redirects per version of your docs.
```

Comment:
An example of how this might be expanded:

Historically we've only supported redirects at the Project level, because we didn't have version level configuration. With the addition of our YAML file, we can now support per version redirects. One use case for this is when you migrate to a new file structure in your upcoming release, but want all your old documentation links to continue working. The work around this issue is being tacked in [issues #xxx]().